### PR TITLE
Enabling Webhook Alerts

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/46341

### Description
Adds the ability to enable webhook notifications for questions

### How to verify
1. Create a valid webhook
2. Go to any saved question and click the share button in the header
3. Select "create an alert" and enable your webhook
4. click test if you'd like to test the alert. Otherwise hit save

### Demo
chrome_QW2pLjZL3Y

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/pulse/components/PulseEditChannels.tsx [hunk 1]
```diff
@@ -1,40 +1,41 @@
 /* eslint "react/prop-types": "warn" */
 import cx from "classnames";
-import { assoc } from "icepick";
+import { assoc, updateIn } from "icepick";
 import { t } from "ttag";
 import _ from "underscore";
 
-import ChannelSetupMessage from "metabase/components/ChannelSetupMessage";
-import Toggle from "metabase/core/components/Toggle";
+// import Toggle from "metabase/core/components/Toggle";
```
Comment (iethree) [inline frontend/src/metabase/pulse/components/PulseEditChannels.tsx:7]: 🗑️

File: frontend/src/metabase/pulse/components/PulseEditChannels.tsx [hunk 2]
```diff
@@ -68,136 +76,74 @@ export const PulseEditChannels = ({
...
+      {notificationChannels.map(notification => (
+        <WebhookChannelEdit
+          key={`webhook-${notification.id}`}
+          user={user}
+          toggleChannel={toggleChannel}
+          channelSpec={channels.http}
+          alert={pulse}
+          notification={notification}
+        />
+      ))}
```
Comment (iethree) [inline frontend/src/metabase/pulse/components/PulseEditChannels.tsx:146]: 😍

File: frontend/src/metabase/query_builder/components/AlertModals/CreateAlertModalContent.jsx [hunk 3]
```diff
@@ -63,7 +63,9 @@ class CreateAlertModalContentInner extends Component {
     this.props.fetchPulseFormInput();
   }
 
-  onAlertChange = alert => this.setState({ alert });
+  onAlertChange = alert => {
+    this.setState({ alert });
+  };
```
Comment (iethree) [inline frontend/src/metabase/query_builder/components/AlertModals/CreateAlertModalContent.jsx:68]: ?

**Review by iethree — APPROVED** [review]
🏆

File: frontend/src/metabase/query_builder/components/AlertModals/CreateAlertModalContent.jsx (hunk 3 above)
Comment (npfitz) [reply frontend/src/metabase/query_builder/components/AlertModals/CreateAlertModalContent.jsx:68]: I modified it to throw a console log in there haha

**Pull request merged**